agent/component/component.go

```diff
@@ -19,12 +19,14 @@ import (
 type Component interface {
 	// Connect to a NATS Server
 	Connect(bCtx *env.BubblyContext) error
+	// Close is used to close any resources, such as connections to NATS, DBs
+	Close()
 	// Subscribe to publications on a given subject
 	Subscribe(bCtx *env.BubblyContext, sub DesiredSubscription) (*nats.Subscription, error)
 	// BulkSubscribe to the component's DesiredSubscriptions
 	BulkSubscribe(bCtx *env.BubblyContext) ([]*nats.Subscription, error)
 	// Publish a publication
-	Publish(bCtx *env.BubblyContext, pub Publication) error
+	// Publish(bCtx *env.BubblyContext, pub Publication) error
 	// Run is the main entrypoint into running the underlying bubbly
 	// processes wrapped by the Component
 	Run(bCtx *env.BubblyContext, agentContext context.Context) error
```

Comment:
I may be forgetting some details, but should this interface not mention some `Publish`/`Request`method interfaces?